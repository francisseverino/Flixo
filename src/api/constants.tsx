export const BASE_URL = 'https://api.themoviedb.org/3';

const API_KEY = process.env.REACT_APP_API_KEY;

//Movie endpoints
export const TRENDING = `/trending/all/week?api_key=${API_KEY}&language=en-US`;
export const NETFLIX_ORIGINALS = `/discover/tv?api_key=${API_KEY}&with_network=213`;
export const TOP_RATED = `/movie/top_rated?api_key=${API_KEY}&language=en-US`;
export const ACTION_MOVIES = `/discover/tv?api_key=${API_KEY}&with_network=28`;
export const COMEDY_MOVIES = `/discover/tv?api_key=${API_KEY}&with_network=35`;
export const HORROR_MOVIES = `/discover/tv?api_key=${API_KEY}&with_network=27`;
export const ROMANCE_MOVIES = `/discover/tv?api_key=${API_KEY}&with_network=10749`;
export const DOCUMENTARIES = `/discover/tv?api_key=${API_KEY}&with_network=99`;

export const requests = {
  trending: `/trending/all/week?api_key=${API_KEY}&language=en-US`,
  netflixOriginals: `/discover/tv?api_key=${API_KEY}&with_network=213`,
  popular: `/movie/popular?api_key=${API_KEY}&language=en-US`,
  topRated: `/movie/top_rated?api_key=${API_KEY}&language=en-US`,
  actionMovies: `/discover/tv?api_key=${API_KEY}&with_network=28`,
  comedyMovies: `/discover/tv?api_key=${API_KEY}&with_network=35`,
  horrorMovies: `/discover/tv?api_key=${API_KEY}&with_network=27`,
  romanceMovies: `/discover/tv?api_key=${API_KEY}&with_network=10749`,
  documentaries: `/discover/tv?api_key=${API_KEY}&with_network=99`,
};
