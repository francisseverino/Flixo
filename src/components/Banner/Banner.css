.banner {
    color: white;
    object-fit: contain;
    width: 100%;
    height: 750px;
    position: relative;
    display: flex;
}

.banner__contents {
    display: flex;
    flex-direction: column;
    margin-left: 30px;
    margin-bottom: 100px;
    justify-content: flex-end;
    width: 35%;
    z-index: 1;

}

.banner__title {
    font-size: 3.2rem;
    font-weight: 800;
}

.banner__description {
    line-height: 1.3;
    font-size: 1.1rem;
    color: rgba(255, 255, 255, 0.9);

    padding: 1.5rem 0;
    /* max-width: 360px; */
}

.banner__button {
    /* display: flex; */
    cursor: pointer;
    color: #fff;
    outline: none;
    font-weight: 700;
    border-radius: 4px;
    border: 1px solid #fff;
    padding-top: 1rem;
    padding-bottom: 1rem;
    padding-left: 2rem;
    padding-right: 2rem;
    margin-right: 1rem;
    background-color: transparent;
    align-items: center;
    justify-content: center;
}

.banner__button:hover {
    color: #000;
    background-color: #e6e6e6;
    transition: all 0.2s;
}

.banner--fadeLeft {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 50%;
    height: 100%;
    background-image: linear-gradient(
        to left,
        transparent,
        rgba(17,17,17,0.6),
        #111
    )
}

.banner--fadeBottom {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 7.4rem;
    background-image: linear-gradient(
        180deg,
        transparent,
        rgba(37,37,37,0.61),
        #111
    )
}