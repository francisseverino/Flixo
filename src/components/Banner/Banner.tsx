import React from 'react';
import request from '../../api/helper';
import { requests } from '../../api/constants';
import { MovieData } from '../../types';
import './Banner.css';

function Banner() {
  const [movie, setMovie] = React.useState<MovieData>();

  React.useEffect(() => {
    request(requests.popular).then(response => {
      const randomIndex = Math.floor(Math.random() * response.results.length - 1);
      setMovie(response.results[randomIndex]);
    });
  }, []);

  const truncate = (str: string, n: number) => {
    return str?.length > n ? str.substr(0, n - 1) + '...' : str;
  };

  return (
    <header
      className='banner'
      style={{
        backgroundSize: 'cover',
        backgroundImage: `url(
            "https://image.tmdb.org/t/p/original/${movie?.backdrop_path}"
        )`,
        backgroundPosition: 'center center',
      }}
    >
      <div className='banner__contents'>
        <h1 className='banner__title'>{movie?.title || movie?.name || movie?.original_name}</h1>

        <h1 className='banner__description'>{truncate(movie?.overview || '', 150)}</h1>
        <div className='banner__buttons'>
          <button className='banner__button'>Play</button>
          <button className='banner__button'>More Info</button>
        </div>
      </div>

      <div className='banner--fadeBottom' />
    </header>
  );
}

export default Banner;
