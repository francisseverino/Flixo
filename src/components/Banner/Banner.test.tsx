import React from 'react';
import Banner from './Banner';
import { render, waitFor } from '@testing-library/react';
import { act } from '@testing-library/react-hooks';
import * as request from '../../api/helper';

describe('Banner component', () => {
  // afterEach(() => {
  //   jest.clearAllMocks();
  // });

  // const requestSpy = jest.spyOn(request, 'default').mockImplementation(() =>
  //   Promise.resolve({
  //     results: [
  //       {
  //         title: 'Frozen',
  //         original_title: 'Original Title Frozen',
  //         backdrop_path: 'mock-path',
  //         name: 'Frozen?',
  //         overview: "A kid's movie which is pretty good for adults but they sing a lot and it gets annoying",
  //       },
  //     ],
  //   })
  // );
  // jest.mock('../../api/helper', () => {
  //   return {
  //     __esModule: true,
  //     default: jest.fn(() =>

  //     ),
  //   };
  // });

  it('renders empty', () => {
    const { asFragment } = render(<Banner />);
    // expect(asFragment()).toMatchSnapshot();
  });

  // it('renders with data', done => {
  //   act(async () => {
  //     const { asFragment } = render(<Banner />);
  //     expect(asFragment()).toMatchSnapshot();
  //     // await waitFor(() => expect(requestSpy).toHaveBeenCalledTimes(1));
  //     expect(requestSpy).toHaveBeenCalled();
  //     // setTimeout(() => {
  //     //   expect(requestSpy).toHaveBeenCalledTimes(1);
  //     //   done();
  //     // }, 100);
  //   });
  // });
});
