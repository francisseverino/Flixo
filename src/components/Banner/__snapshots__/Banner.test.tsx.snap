// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders 1`] = `
<DocumentFragment>
  <header
    class="banner"
    style="background-size: cover; background-position: center center;"
  >
    <div
      class="banner__contents"
    >
      <h1
        class="banner__title"
      />
      <h1
        class="banner__description"
      />
      <div
        class="banner__buttons"
      >
        <button
          class="banner__button"
        >
          Play
        </button>
        <button
          class="banner__button"
        >
          More Info
        </button>
      </div>
    </div>
    <div
      class="banner--fadeBottom"
    />
  </header>
</DocumentFragment>
`;

exports[`Banner component renders empty 1`] = `
<DocumentFragment>
  <header
    class="banner"
    style="background-size: cover; background-position: center center;"
  >
    <div
      class="banner__contents"
    >
      <h1
        class="banner__title"
      />
      <h1
        class="banner__description"
      />
      <div
        class="banner__buttons"
      >
        <button
          class="banner__button"
        >
          Play
        </button>
        <button
          class="banner__button"
        >
          More Info
        </button>
      </div>
    </div>
    <div
      class="banner--fadeBottom"
    />
  </header>
</DocumentFragment>
`;

exports[`Banner component renders with data 1`] = `
<DocumentFragment>
  <header
    class="banner"
    style="background-size: cover; background-position: center center;"
  >
    <div
      class="banner__contents"
    >
      <h1
        class="banner__title"
      />
      <h1
        class="banner__description"
      />
      <div
        class="banner__buttons"
      >
        <button
          class="banner__button"
        >
          Play
        </button>
        <button
          class="banner__button"
        >
          More Info
        </button>
      </div>
    </div>
    <div
      class="banner--fadeBottom"
    />
  </header>
</DocumentFragment>
`;
